src/main/java/Deadline.java

```diff
@@ -0,0 +1,75 @@
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+public class Deadline extends Task {
+    private String byString;
+    private LocalDate byDate;
+    private LocalDateTime byDateTime;
+
+    public Deadline(String description, String by) {
+        super(description, TaskType.DEADLINE);
+        this.byString = by;
+        parseDateTime(by);
+    }
+
+    /**
+     * Attempts to parse the deadline string as a date/time.
+     */
+    private void parseDateTime(String by) {
+        this.byDateTime = DateTimeParser.parseDateTime(by);
+
+        if (this.byDateTime == null) {
+            this.byDate = DateTimeParser.parseDate(by);
+        }
+    }
+
+    public String getBy() {
```

Comment:
```suggestion
    public String getByString() {
```
Perhaps you would like to match the method name with the return variable name? I noticed the same with Event.java